mm2src/coins/z_coin/storage/walletdb/wasm/storage.rs

```diff
@@ -0,0 +1,1494 @@
+        //      account, up to itself.
+        //    - Spent notes accumulate the values of all notes in the transaction they were
+        //      spent in, up to itself.
+        //
+        // 2) Select all unspent notes in the desired account, along with their running sum.
+        //
+        // 3) Select all notes for which the running sum was less than the required value, as
+        //    well as a single note for which the sum was greater than or equal to the
+        //    required value, bringing the sum of all selected notes across the threshold.
+        //
+        // 4) Match the selected notes against the witnesses at the desired height.
+        let ticker = self.ticker.clone();
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+
+        // Received notes
+        let received_notes_table = db_transaction.table::<WalletDbReceivedNotesTable>().await?;
+        let index_keys = MultiIndex::new(WalletDbReceivedNotesTable::TICKER_ACCOUNT_INDEX)
+            .with_value(&ticker)?
+            .with_value(account.0.to_bigint().unwrap())?;
+        let maybe_notes = received_notes_table.get_items_by_multi_index(index_keys).await?;
+
+        // Transactions
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let txs_table = db_transaction.table::<WalletDbTransactionsTable>().await?;
+        let mut maybe_txs = txs_table
+            .cursor_builder()
+            .only("ticker", ticker.clone())?
+            .bound("block", 0u32, u32::from(anchor_height))
+            .open_cursor(WalletDbTransactionsTable::TICKER_BLOCK_INDEX)
+            .await?;
+        let mut txs = vec![];
+        while let Some((id, ts)) = maybe_txs.next().await? {
+            txs.push((id, ts))
+        }
+
+        // Sapling Witness
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let witness_table = db_transaction.table::<WalletDbSaplingWitnessesTable>().await?;
+        let index_keys = MultiIndex::new(WalletDbSaplingWitnessesTable::TICKER_BLOCK_INDEX)
+            .with_value(&ticker)?
+            .with_value(u32::from(anchor_height))?;
+        let witnesses = witness_table.get_items_by_multi_index(index_keys).await?;
+
+        let mut running_sum = 0;
+        let mut notes = vec![];
+        for (id_note, note) in &maybe_notes {
+            let value = note.value.clone().to_i64().expect("price is too large");
+            if note.spent.is_none() {
+                running_sum += value;
+                notes.push((id_note, value, note, running_sum));
+            }
+        }
+
+        let mut final_note = vec![];
+        for (id_note, value, note, running_sum) in &notes {
+            if let Some(_tx_block) = txs.iter().find(|(id_tx, _tx)| *id_tx == note.tx) {
+                final_note.push((id_note, value, note, running_sum));
+            }
+        }
```

Comment:
Same as `maybe_notes`. These iterations/mutatings are duplicated a lot, I will leave the rest to you instead of pointing all of them.

Reply:
all done